### PR TITLE
feat: cmaf investigation

### PR DESCRIPTION
**Pull request by lfordyce**

### CODE REVIEW

File: packager/live_packager_test.cc [hunk 1]
```diff
@@ -1201,7 +1201,22 @@ INSTANTIATE_TEST_CASE_P(
         LivePackagerTestCase{
             5, "audio/en/init.mp4", LiveConfig::EncryptionScheme::SAMPLE_AES,
             LiveConfig::OutputFormat::TS, LiveConfig::TrackType::AUDIO,
-            "audio/en/%05d.m4s", false}));
+            "audio/en/%05d.m4s", false},
```
Comment (chenda6) [inline packager/live_packager_test.cc:1204]: Could we enable the `compare_samples` flag on these cases for a more thorough validation? (will likely need to update `ReadExpectedData()` to accommodate this)

Comment (lfordyce) [reply packager/live_packager_test.cc:1204]: What should we compare against? Should we write out packaged samples and compare against those?

Comment (chenda6) [reply packager/live_packager_test.cc:1204]: I was thinking that we could compare against the source samples since packaging should not alter them

Comment (lfordyce) [reply packager/live_packager_test.cc:1204]: Unfortunately that doesn't work with how the fragments are currently captured. From SIM, the media fragments don't capture the segment type boxes (eg `styp`). https://github.com/Pluto-tv/service-ingest-media/blob/feature/TRANS-5884-cmaf-service/internal/cmaf/mp4_reader.go#L106. So the fragments that are eventually passed into shaka only contain the `moof` and `mdat` boxes.

Shaka will add in the these box types as well as `sidx` box types for each fragment.

Comment (chenda6) [reply packager/live_packager_test.cc:1204]: Could you elaborate on what doesn't work?  The fact that Shaka can package the segments indicates that it is able to parse the samples.  So the the parsers used in this test should work.

File: packager/live_packager_test.cc [hunk 2]
```diff
@@ -1499,4 +1534,33 @@ INSTANTIATE_TEST_CASE_P(
...
+  SegmentData media_seg(segment_buffer.data(), segment_buffer.size());
+
+  FullSegmentBuffer out;
+
+  LiveConfig live_config;
+  live_config.format = LiveConfig::OutputFormat::VTTMP4;
+  live_config.track_type = LiveConfig::TrackType::TEXT;
+  live_config.protection_scheme = LiveConfig::EncryptionScheme::NONE;
+  live_config.segment_number = 343138171;
+  live_config.timed_text_decode_time = (154412176500000 / 90000) * 1000;
```
Comment (chenda6) [inline packager/live_packager_test.cc:1552]: Could you please leave a comment here to indicate the intent of this test, i.e., to ensure a large decode time (> 32 bits) can be properly handled?

**Review by chenda6 — APPROVED** [review]

**Pull request merged**